### PR TITLE
PHP packages & extensions ZHF

### PR DESCRIPTION
**Pull request by Ma27**
Marks a bunch of things as broken and fixes `psysh` and `php84Extensions.soap`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -663,7 +663,7 @@ in {
           # Some tests are causing issues in the Darwin sandbox with issues
           # such as
           #   Unknown: php_network_getaddresses: getaddrinfo for localhost failed: nodename nor servname provided
-          doCheck = !stdenv.hostPlatform.isDarwin;
+          doCheck = !stdenv.hostPlatform.isDarwin && lib.versionOlder php.version "8.4";
```
Comment (Ma27) [inline pkgs/top-level/php-packages.nix:666]: See the commit message for the rationale. If somebody has a better solution, I'm all for it.

File: pkgs/development/php-packages/psysh/default.nix [hunk 2]
```diff
@@ -19,7 +19,7 @@ let
   composerLock = fetchurl {
     name = "composer.lock";
     url = "https://github.com/bobthecow/psysh/releases/download/v${version}/composer-v${version}.lock";
-    hash = "sha256-ur6mzla3uXeFL6aEHAPdpxGdvcgzOgTLW/CKPbNqeCg=";
+    hash = "sha256-PQDWShzvTY8yF+OUPVJAV0HMx0/KnA03TDhZUM7ppXw=";
```
Comment (Ma27) [inline pkgs/development/php-packages/psysh/default.nix:22]: cc @bobthecow did you force-push the 0.12.4 tag back then?

**Review by patka-123 — APPROVED** [review]
Looks good and makes sense. Builds fine, as expected.

Thanks!

File: pkgs/development/php-packages/psysh/default.nix (hunk 2 above)
Comment (Ma27) [reply pkgs/development/php-packages/psysh/default.nix:22]: Hmm on a second look it seems as if 0.12 never built, so perhaps the FOD hash was just wrong from the beginning.
Never mind then.

**Review by piotrkwiecinski — APPROVED** [review]

**Comment by drupol** [comment]
hey @piotrkwiecinski ! Long time no see! All good?

**Review by drupol — APPROVED** [review]

**Comment by piotrkwiecinski** [comment]
hi @drupol . All good. Live happened. I'm slowly getting back to contributions.

**Pull request merged**